syntax = "proto3";

package sailbot.msg;

option cc_enable_arenas = true;

message SailCmd {
  float vel = 11;
}

message RudderCmd {
  float vel = 11;
}

message Vector3f {
  float x = 1;
  float y = 2;
  float z = 3;
}

message Quaternion {
  float w = 1;
  float x = 2;
  float y = 3;
  float z = 4;
}

message InternalBoatState {
  float sail = 1;
  float rudder = 2;
}

message BoatState {
  Vector3f pos = 1;
  Quaternion orientation = 2;
  Vector3f vel = 3;
  Vector3f omega = 4;
  InternalBoatState internal = 5;
}
